### PR TITLE
Change Tokens update to 2.2

### PR DESCRIPTION
**Pull request by guardrex**
Addresses #5495 

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/change-tokens?view=aspnetcore-2.1&branch=pr-en-us-13099)

**No rush at all on review. Post-holiday/weekend is cool.**

### CODE REVIEW

File: aspnetcore/fundamentals/change-tokens.md [hunk 1]
```diff
@@ -2,169 +2,187 @@
 title: Detect changes with change tokens in ASP.NET Core
 author: guardrex
 description: Learn how to use change tokens to track changes.
+monikerRange: '>= aspnetcore-2.1'
 ms.author: riande
-ms.date: 11/10/2017
+ms.date: 07/03/2017
```
Comment (scottaddie) [inline aspnetcore/fundamentals/change-tokens.md]:
```suggestion
ms.date: 07/03/2019
```

File: aspnetcore/fundamentals/change-tokens.md [hunk 2]
```diff
@@ -2,169 +2,187 @@
...
-[!code-csharp[](change-tokens/sample/Startup.cs?name=snippet2)]
+[!code-csharp[](change-tokens/samples/2.x/SampleApp/Startup.cs?name=snippet2)]
 
 `config.GetReloadToken()` provides the token. The callback is the `InvokeChanged` method:
 
-[!code-csharp[](change-tokens/sample/Startup.cs?name=snippet3)]
+[!code-csharp[](change-tokens/samples/2.x/SampleApp/Startup.cs?name=snippet3)]
 
-The `state` of the callback is used to pass in the `IHostingEnvironment`. This is useful to determine the correct *appsettings* configuration JSON file to monitor, *appsettings.&lt;Environment&gt;.json*. File hashes are used to prevent the `WriteConsole` statement from running multiple times due to multiple token callbacks when the configuration file has only changed once.
+The `state` of the callback is used to pass in the `IHostingEnvironment`. This is useful to determine the correct *appsettings* configuration JSON file to monitor, *appsettings.{ENVIRONMENT}.json*. File hashes are used to prevent the `WriteConsole` statement from running multiple times due to multiple token callbacks when the configuration file has only changed once.
```
Comment (scottaddie) [inline aspnetcore/fundamentals/change-tokens.md]: This 2nd sentence reads poorly. Seems like something is needed after "monitor, ".

File: aspnetcore/fundamentals/change-tokens.md [hunk 3]
```diff
@@ -2,169 +2,187 @@
...
+1. A change token is obtained from the file provider with [IFileProviders.Watch](xref:Microsoft.Extensions.FileProviders.IFileProvider.Watch*). The token's callback is triggered when the file is modified.
+1. The file content is cached with a [sliding expiration](xref:Microsoft.Extensions.Caching.Memory.MemoryCacheEntryOptions.SlidingExpiration) period. The change token is attached with [MemoryCacheEntryExtensions.AddExpirationToken](xref:Microsoft.Extensions.Caching.Memory.MemoryCacheEntryExtensions.AddExpirationToken*) to evict the cache entry if the file changes while it's cached.
+
+In the following example code, files are stored in the content root of the app. [IHostingEnvironment.ContentRootFileProvider](xref:Microsoft.AspNetCore.Hosting.IHostingEnvironment.ContentRootFileProvider) is used to obtain an <xref:Microsoft.Extensions.FileProviders.IFileProvider> pointing at the app's <xref:Microsoft.AspNetCore.Hosting.IHostingEnvironment.ContentRootPath>. The `filePath` is obtained with [IFileInfo.PhysicalPath](xref:Microsoft.Extensions.FileProviders.IFileInfo.PhysicalPath).
```
Comment (scottaddie) [inline aspnetcore/fundamentals/change-tokens.md]:
```suggestion
In the following example, files are stored in the app's content root. [IHostingEnvironment.ContentRootFileProvider](xref:Microsoft.AspNetCore.Hosting.IHostingEnvironment.ContentRootFileProvider) is used to obtain an <xref:Microsoft.Extensions.FileProviders.IFileProvider> pointing at the app's <xref:Microsoft.AspNetCore.Hosting.IHostingEnvironment.ContentRootPath>. The `filePath` is obtained with [IFileInfo.PhysicalPath](xref:Microsoft.Extensions.FileProviders.IFileInfo.PhysicalPath).
```

File: aspnetcore/fundamentals/change-tokens.md (hunk 1 above)
Comment (guardrex) [reply aspnetcore/fundamentals/change-tokens.md]: 🤦‍♂

**Comment by guardrex** [comment]
@scottaddie Try it now.

*It isn't 2017?*

Hummm 🤔 ...... I seem to have missed a few years somewhere.

**Pull request merged**